src/state/queries/resolve-identity.ts

```diff
@@ -0,0 +1,27 @@
+import {LRU} from './direct-fetch-record'
+
+// TODO: lift this into direct fetch to share cache
+const serviceCache = new LRU<`did:${string}`, string>()
+
+export async function resolvePdsUrl(did: `did:${string}`) {
+  return await serviceCache.getOrTryInsertWith(did, async () => {
```

Comment:
I regret not making a higher order function that makes a function LRU cached... You don't need to do that in this PR tho.